### PR TITLE
EVG-5932: set-module for commit queue CLI

### PR DESCRIPTION
**Pull request by ybrill**

### CODE REVIEW

File: operations/commit_queue.go [hunk 1]
```diff
@@ -150,6 +151,45 @@ func mergeCommand() cli.Command {
...
+		Usage: "update or add module to an existing merge patch",
+		Flags: mergeFlagSlices(addLargeFlag(), addPatchIDFlag(), addModuleFlag(), addYesFlag(
+			cli.StringFlag{
+				Name:  joinFlagNames(refFlagName, "r"),
+				Usage: "merge branch `REF`",
+				Value: "HEAD",
+			},
+		)),
+		Before: mergeBeforeFuncs(
+			setPlainLogger,
```
Comment (tychoish) [inline operations/commit_queue.go]: do we want to plain log?

File: operations/commit_queue.go [hunk 2]
```diff
@@ -150,6 +151,45 @@ func mergeCommand() cli.Command {
...
+			conf, err := NewClientSettings(c.Parent().String(confFlagName))
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			ac, rc, err := conf.getLegacyClients()
+			if err != nil {
+				return errors.Wrap(err, "problem accessing evergreen service")
+			}
+
+			return params.addModule(ac, rc)
```
Comment (tychoish) [inline operations/commit_queue.go]: we should wrap this error?

File: operations/commit_queue.go [hunk 3]
```diff
@@ -245,6 +285,62 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
...
+	module      string
+	ref         string
+	large       bool
+	skipConfirm bool
+}
+
+func (p *moduleParams) addModule(ac *legacyClient, rc *legacyClient) error {
+	proj, err := rc.GetPatchedConfig(p.patchID)
+	if err != nil {
+		return err
```
Comment (tychoish) [inline operations/commit_queue.go]: wrap?

File: operations/commit_queue.go [hunk 4]
```diff
@@ -245,6 +285,62 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
...
+		}
+
+		if !confirm("This is a summary of the patch to be submitted. Continue? (y/n):", true) {
+			return nil
+		}
+	}
+
+	err = ac.UpdatePatchModule(p.patchID, p.module, diffData.fullPatch, diffData.base)
+	if err != nil {
+		return err
```
Comment (tychoish) [inline operations/commit_queue.go]: wrap

File: operations/commit_queue.go [hunk 5]
```diff
@@ -245,6 +285,62 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
...
+	}
+
+	if err = validatePatchSize(diffData, p.large); err != nil {
+		return err
+	}
+
+	if !p.skipConfirm {
+		grip.Infof("Using branch %s for module %s", moduleBranch, p.module)
+		if diffData.patchSummary != "" {
+			grip.Info(diffData.patchSummary)
```
Comment (tychoish) [inline operations/commit_queue.go]: bury the conditional in an infoWhen

File: operations/commit_queue.go [hunk 6]
```diff
@@ -245,6 +285,62 @@ func (p *mergeParams) uploadMergePatch(conf *ClientSettings, ac *legacyClient) e
...
 
+type moduleParams struct {
+	patchID     string
+	module      string
+	ref         string
+	large       bool
+	skipConfirm bool
+}
+
+func (p *moduleParams) addModule(ac *legacyClient, rc *legacyClient) error {
```
Comment (tychoish) [inline operations/commit_queue.go]: I might break this apart. right now the confirms would make it imposslbe to call this code as a test.

File: operations/commit_queue.go (hunk 1 above)
Comment (ybrill) [reply operations/commit_queue.go]: done

File: operations/commit_queue.go (hunk 2 above)
Comment (ybrill) [reply operations/commit_queue.go]: The thinking of not wrapping these errors is to pass the message from the server to the user unchanged. (I think @bsamek liked it this way)

File: operations/commit_queue.go (hunk 5 above)
Comment (ybrill) [reply operations/commit_queue.go]: done.

File: operations/commit_queue.go (hunk 6 above)
Comment (ybrill) [reply operations/commit_queue.go]: done. I would argue, however, the confirms are surmountable by passing `p.skipConfirm == true`

Comment (ybrill) [reply operations/commit_queue.go]: I find shelling out to git to be an issue, though.

File: operations/commit_queue.go (hunk 2 above)
Comment (tychoish) [reply operations/commit_queue.go]: In these cases use WithStack()... the problem is that the errors you're passing around are local to your environment and only mirror things on the server side. 

Wrapping errors makes it possible to figure out what happened when you actually get an error and to be able to meaningfully understand where an error occurred. The only reason to not wrap an error is if you need to introspect the error, and if that's the case errors.Cause() unrwaps errors (and really error introspection is risky enough, that we should avoid it when possible, which makes the case for "not wrapping" to be minimal.

**Pull request merged**